service_restaurant/src/main/kotlin/org/thechance/service_restaurant/usecase/GetRestaurantsUseCaseImp.kt

```diff
@@ -0,0 +1,11 @@
+package org.thechance.service_restaurant.usecase
+
+import org.koin.core.annotation.Single
+import org.thechance.service_restaurant.api.usecases.GetRestaurantsUseCase
+import org.thechance.service_restaurant.usecase.gateway.RestaurantGateway
+
+@Single
+class GetRestaurantsUseCaseImp(private val restaurant: RestaurantGateway) : GetRestaurantsUseCase {
```

Comment:
A better naming for restaurant here would be restaurants since you can call getRestaurants on it, so you don't get list of restaurants from single restaurant.

Reply:
@Shehab-m no restaurant here represents a Getway /DaraSourse not a List of restaurants.

Reply:
Yes, you are right. I was just talking for readability purposes. Maybe it could be just gateway or not that big of a deal i was just saying my opinion.

Reply:
@Shehab-m So When I have another getaway, what should I name it? This is possible because we will have categories and other features. By the way, this branch is for adding dependency injection, not CRUD operations!!!